cli/src/semgrep/run_scan.py

```diff
@@ -422,7 +422,12 @@ def run_scan(
         metrics.add_project_url(project_url)
         metrics.add_integration_name(environ.get("SEMGREP_INTEGRATION_NAME"))
         metrics.add_configs(configs)
-        metrics.add_engine_type(engine_type)
+        metrics.add_engine_config(
+            engine_type,
+            ("all" if allow_untrusted_validators else "pro_rules")
+            if run_secrets and not disable_secrets_validation
+            else None,
```

Comment:
To be honest, if I didn't have some secrets context, I think after reading this multiple times I still would not understand how to interpret this wrt answering the question "Did secrets run?". Would like to simplify the type

Reply:
I am not sure exactly what simplification you would like: ultimately a `None` for `secrets_config` would mean secrets was not run. `Some { secrets_origins = "..." }` would correspond to secrets running with that option for the origins. I think what the origins mean is not obvious, but that seems a bit distinct.

Reply:
I think the nicest thing would be to have a class for the secrets config type which is also passed to the semgrep-pro engine, and which implements `str`. Don't know how practical this is.

Reply:
Just changed to an extensible sum.